hazelcast/include/hazelcast/client/protocol/ClientMessage.h

```diff
@@ -795,6 +877,173 @@ class HAZELCAST_API ClientMessage
+        skip_frame();
+
+        auto column_type_ids = get<std::vector<int32_t>>();
+
+        using column = std::vector<boost::any>;
+
+        std::vector<column> columns;
+        std::vector<sql::sql_column_type> column_types;
+
+        for (auto column_type_id : column_type_ids) {
+            auto column_type = static_cast<sql::sql_column_type>(column_type_id);
+            column_types.push_back(column_type);
+
+            switch (column_type) {
+                case sql::sql_column_type::varchar:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<std::string>>>()));
+                    break;
+                case sql::sql_column_type::boolean:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<bool>>>()));
+                    break;
+                case sql::sql_column_type::tinyint:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<byte>>>()));
+                    break;
+                case sql::sql_column_type::smallint:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<int16_t>>>()));
+                    break;
+                case sql::sql_column_type::integer:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<int32_t>>>()));
+                    break;
+                case sql::sql_column_type::bigint:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<int64_t>>>()));
+                    break;
+                case sql::sql_column_type::real:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<float>>>()));
+                    break;
+                case sql::sql_column_type::double_:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<double>>>()));
+                    break;
+                case sql::sql_column_type::date:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<local_date>>>()));
+                    break;
+                case sql::sql_column_type::time:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<local_time>>>()));
+                    break;
+                case sql::sql_column_type::timestamp:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<local_date_time>>>()));
+                    break;
+                case sql::sql_column_type::timestamp_with_timezone:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<offset_date_time>>>()));
+                    break;
+                case sql::sql_column_type::decimal:
+                    columns.emplace_back(to_vector_of_any(get<std::vector<boost::optional<big_decimal>>>()));
+                    break;
+                case sql::sql_column_type::null: {
```

Comment:
Where is ROW, OBJECT or JSON?

Reply:
Added OBJECT. JSON is to be added in later ticket. 
What is ROW?

Reply:
ROW is for nested support. It is experimental. I think we can skip it for nonjava